agent/connect/ca/provider_vault.go

```diff
@@ -381,19 +384,47 @@ func (v *VaultProvider) setupIntermediatePKIPath() error {
 				Config:      mountConfig,
 			})
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to mount intermediate PKI backend: %w", err)
 			}
+			// Required to determine if we should tune the mount
+			// if the VaultProvider is ever reconfigured.
+			v.isConsulMountedIntermediate = true
+
+		} else if err == ErrBackendNotInitialized {
+			// If this is the first time calling setupIntermediatePKIPath, the backend
+			// will not have been initialized. Since the mount is ready we can suppress
+			// this error.
 		} else {
-			return err
+			return fmt.Errorf("unexpected error while fetching intermediate CA: %w", err)
 		}
 	} else {
-		err := v.tuneMountNamespaced(v.config.IntermediatePKINamespace, v.config.IntermediatePKIPath, &mountConfig)
-		if err != nil {
-			v.logger.Warn("Could not update intermediate PKI mount settings", "path", v.config.IntermediatePKIPath, "error", err)
+		// If Consul was responsible for mounting the intermediate PKI path
+		// we should update the mount with any new config.
+		if v.isConsulMountedIntermediate {
+			// This codepath requires the Vault policy:
+			//
+			//   path "/sys/mounts/<intermediate_pki_path>/tune" {
+			//     capabilities = [ "update" ]
+			//   }
+			//
+			err := v.tuneMountNamespaced(v.config.IntermediatePKINamespace, v.config.IntermediatePKIPath, &mountConfig)
+			if err != nil {
+				v.logger.Warn("Intermediate PKI path was mounted by Consul but could not be tuned",
+					"namespace", v.config.IntermediatePKINamespace,
+					"path", v.config.IntermediatePKIPath,
+					"error", err,
+				)
+			}
+
+		} else {
+			v.logger.Info("Found existing Intermediate PKI path mount",
+				"namespace", v.config.IntermediatePKINamespace,
+				"path", v.config.IntermediatePKIPath,
+			)
 		}
```

Comment:
Since tuning the mount is only relevant for Consul-managed, I added a conditional to skip since otherwise you'd see an ominous WARN in the logs on startup.

This isn't the cleanest way to determine if the provider is "Consul-managed" or "Vault-managed" but introducing explicit configuration requires some more consideration around default/missing values.

Reply:
Tuning this mount really only matters if the mount's `MaxLeaseTTL` is lower than the `IntermediateCertTTL`. Otherwise, tuning the config has no effect, as the [intermediate CA cert will still be issued with the requested TTL](https://github.com/hashicorp/consul/blob/b1013e06b56f522cbb96d24eb08a7dbd9d2c2d05/agent/connect/ca/provider_vault.go#LL587C39-L587C39).

I think I heard, though haven't tested, that if you ask Vault to generate a cert (i.e., intermediate CA cert) with a TTL greater than `MaxLeaseTTL`, the operation will still succeed, the TTL will just be truncated to `MaxLeaseTTL`. So everything will still *work*, it's just that the intermediate CA cert will expire sooner than configured. (Unless there's something about our "attempt to renew at 50% of lifetime" math that breaks because it depends on `IntermediateCertTTL` rather than the TTL of the actual issuer cert.)

Reply:
I think its somewhat short: https://developer.hashicorp.com/vault/docs/concepts/tokens#the-general-case -- 32 days.

You'll definitely want to read the cert's actual validity period rather than assuming the TTL is OK, not sure if that's useful in this PR or a follow-up though.

Also, I think in 1.11+, we've started adding a warning in this event, but obviously that doesn't help Vault <= 1.10...